contracts/src/token/erc20/extensions/flash_mint.rs

```diff
@@ -0,0 +1,318 @@
+    #[derive(Debug)]
+    #[allow(missing_docs)]
+    error ERC3156ExceededMaxLoan(uint256 max_loan);
+
+    /// Indicate  an  error related to an invalid flash loan receiver.
+    /// The receiver does not implement the required `onFlashLoan` function.
+    #[derive(Debug)]
+    #[allow(missing_docs)]
+    error ERC3156InvalidReceiver(address receiver);
+}
+
+/// A FlashMint error.
+#[derive(SolidityError, Debug)]
+pub enum Error {
+    /// Indicate an error related to an unsupported loan token.
+    /// This occurs when the specified token cannot be used for loans.
+    UnsupportedToken(ERC3156UnsupportedToken),
+    /// Indicate an error related to the loan amount exceeds the maximum.
+    /// The requested amount is higher than the allowed loan for this token
+    /// max_loan.
+    ExceededMaxLoan(ERC3156ExceededMaxLoan),
+    /// Indicate an error related to an invalid flash loan receiver.
+    /// The receiver does not implement the required `onFlashLoan` function.
+    InvalidReceiver(ERC3156InvalidReceiver),
+    /// Error type from [`Erc20`] contract [`erc20::Error`].
+    Erc20(erc20::Error),
+}
+
+pub use borrower::IERC3156FlashBorrower;
+#[allow(missing_docs)]
+mod borrower {
+    use stylus_sdk::stylus_proc::sol_interface;
+
+    sol_interface! {
+        /// Interface that must be implemented by smart contracts
+        /// in order to borrow ERC-3156 flashloan .
+        interface IERC3156FlashBorrower {
+            /// Handles the receipt of a flash loan.
+            /// This function is called after the loan amount has been transferred to the borrower.
+            ///
+            /// To indicate successful handling of the flash loan, this function should return
+            /// the `keccak256` hash of "ERC3156FlashBorrower.onFlashLoan".
+            ///
+            /// # Arguments
+            ///
+            /// * `initiator` - The address which initiated the flash loan.
+            /// * `token` - The address of the token being loaned (loan currency).
+            /// * `amount` - The amount of tokens lent in the flash loan.
+            /// * `fee` - The additional fee to repay with the flash loan amount.
+            /// * `data` - Arbitrary data structure, intended to contain user-defined parameters.
+            #[allow(missing_docs)]
+            function onFlashLoan(
+                address initiator,
+                address token,
+                uint256 amount,
+                uint256 fee,
+                bytes calldata data
+            ) external returns (bytes32);
+        }
+    }
+}
+
+/// State of the [`Erc20FlashMint`] Contract.
+#[storage]
+pub struct Erc20FlashMint {
+    /// Fee applied when doing flash loans.
+    pub flash_fee_amount: StorageU256,
+    /// Receiver address of the flash fee.
+    pub flash_fee_receiver_address: StorageAddress,
+}
+
+/// Extension of [`Erc20`] that allows token holders to destroy both
```

Comment:
I believe that this is not a right docs for this extension. Check [this](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v5.1.0/contracts/interfaces/IERC3156FlashLender.sol) out.